l3build-aux.lua

```diff
@@ -57,69 +75,89 @@ function setepoch()
       .. os_concat
 end
 
-local function getscriptname()
+---Returns the script name depending on the calling sequence.
+---`l3build ...` -> full path of `l3build.lua` in the TDS
+---When called via `texlua l3build.lua ...`, `l3build.lua` is resolved to either 
+---`./l3build.lua` or the full path of `l3build.lua` in the TDS.
+---`texlua l3build.lua` -> `/Library/TeX/texbin/l3build.lua` or `./l3build.lua`
+---@return string
+local function get_script_name()
   if match(arg[0], "l3build$") or match(arg[0], "l3build%.lua$") then
     return lookup("l3build.lua")
   else
-    return arg[0]
+    return arg[0] -- Why no lookup here? 
   end
 end
 
--- Do some subtarget for all modules in a bundle
-function call(dirs, target, opts)
-  -- Turn the option table into a string
-  local opts = opts or options
-  local s = ""
+-- Performs the task named target given modules in a bundle.
+---A module is the path of a directory relative to the main one.
+---Uses `run` to launch a command.
+---@param modules table List of modules.
+---@param target string
+---@param opts table
+---@return number 0 on proper termination, a non 0 error code otherwise.
+---@see many places, including latex2e/build.lua
+---@usage Public
+function call(modules, target, opts)
+  -- Turn the option table into a CLI option string
+  opts = opts or options
+  local cli_opts = ""
   for k,v in pairs(opts) do
-    if k ~= "names" and k ~= "target" then -- Special cases
-      local t = option_list[k] or { }
-      local arg = ""
+    if k ~= "names" and k ~= "target" then -- Special cases, to be removed!!!
+      local t = option_list[k] or {}
+      local value = ""
       if t["type"] == "string" then
-        arg = arg .. "=" .. v
-      end
-      if t["type"] == "table" then
+        value = value .. "=" .. v
+      elseif t["type"] == "table" then
         for _,a in pairs(v) do
-          if arg == "" then
-            arg = "=" .. a -- Add the initial "=" here
+          if value == "" then
+            value = "=" .. a -- Add the initial "=" here
           else
-            arg = arg .. "," .. a
+            value = value .. "," .. a
           end
         end
       end
-      s = s .. " --" .. k .. arg
+      cli_opts = cli_opts .. " --" .. k .. value
     end
   end
-  if opts["names"] then
-    for _,v in pairs(opts["names"]) do
-      s = s .. " " .. v
+  if opts.names then
+    for _, name in pairs(opts.names) do
+      cli_opts = cli_opts .. " " .. name
     end
   end
-  local scriptname = getscriptname()
-  for _,i in ipairs(dirs) do
-    local text = " for module " .. i
```

Comment:
I don't see where this text gets moved to ...

Reply:
one line 134